### PR TITLE
Bump to using JDK 19 as the latest JDK in CI

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: .github/matrix-jvm-tests.json [hunk 1]
```diff
@@ -13,8 +13,8 @@
   "os-name": "ubuntu-latest"
 }
 , {
-  "name": "18",
-  "java-version": 18,
+  "name": "19",
+  "java-version": 19,
```
Comment (gsmet) [inline .github/matrix-jvm-tests.json:17]: It probably won't work because of Gradle.

Comment (geoand) [reply .github/matrix-jvm-tests.json:17]: That's why I have the PR in draft. I'm waiting for CI on my fork

Comment (geoand) [reply .github/matrix-jvm-tests.json:17]: Great, thanks!

Comment (geoand) [reply .github/matrix-jvm-tests.json:17]: Indeed CI fails on some Gradle stuff. Seems we need to wait for Gradle 7.6 for this bump.

**Comment by geoand** [comment]
Seems like there is an actual issue.
I'll take a quick look

**Comment by geoand** [comment]
Seems like we have [not](https://github.com/quarkusio/quarkus/pull/29508#issuecomment-1360145007) bumped Gradle to 7.6 yet

**Comment by gsmet** [comment]
I rebased this one as we updated to Gradle 8, which should support Java 19.

**Comment by geoand** [comment]
Great, I had forgot about it!

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
The JDK 19 build succeeded so merging!

**Comment by geoand** [comment]
🎉

**Pull request merged**